File: app/src/main/java/kisan/com/fabproject/ForcastAdapter.java
package kisan.com.fabproject;


import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import java.util.List;

/**
 * Created by hp on 07-12-2018.
 */

public class ForcastAdapter extends RecyclerView.Adapter<ForcastAdapter.ViewHolder> {
    List<ForcastModel> list;
    public ForcastAdapter(List<ForcastModel> list){
    this.list = list;
    }
    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new ViewHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.forcast_list,parent,false));
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
       holder.condition.setText(list.get(position).getCondition());
       holder.date.setText(list.get(position).getDate());
       holder.temp.setText(list.get(position).getMinT() + "/" + list.get(position).getMaxT());
       switch (list.get(position).getConditionCode()){
           case 1003:
               holder.imageView.setImageResource(R.drawable.clouds);
               break;
           default:
               holder.imageView.setImageResource(R.drawable.clear);

       }
    }



    @Override
    public int getItemCount() {
        return list.size();
    }

    class ViewHolder extends RecyclerView.ViewHolder{
        TextView date;
        TextView temp;
        TextView condition;
        ImageView imageView;
        public ViewHolder(View itemView) {
            super(itemView);
            init(itemView);
        }

        private void init(View itemView) {
            date = itemView.findViewById(R.id.date);
            temp = itemView.findViewById(R.id.temp);
            condition = itemView.findViewById(R.id.condition);
            imageView = itemView.findViewById(R.id.imageView);
        }
    }
}

File: app/src/main/java/kisan/com/fabproject/ForcastAdapter.kt
package kisan.com.fabproject


import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import kotlinx.android.synthetic.main.forcast_list.view.*
import java.text.SimpleDateFormat
import java.time.LocalDate
import java.time.format.DateTimeFormatter

/**
 * Created by hp on 07-12-2018.
 */

class ForcastAdapter(internal var list: List<ForcastModel>) : RecyclerView.Adapter<ForcastAdapter.ViewHolder>() {


    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        return ViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.forcast_list, parent, false))
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {


        holder.itemView.date.text = getDate(list[position].date)
        holder.itemView.temp.text = list[position].maxT.toFloat().toInt().toString() + " C"

    }

    private fun getDate(dat: String): String {
        val date = SimpleDateFormat("yyyy-MM-dd").parse(dat)
       return  SimpleDateFormat("EEEE").format(date)
    }


    override fun getItemCount(): Int {
        return list.size
    }

     inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView)
}
